### ISSUE TITLE
`setPort()` and/or `getPort()` is broken on AMQPConnection

### ISSUE THREAD

**Issue by pkruithof** [pdezwart/php-amqp#188]
On 1.7.0-alpha1:

```php
$conn = new \AMQPConnection(
    [
        'host' => 'localhost',
        'port' => 1234,
        'login' => 'guest',
        'password' => 'guest',
        'vhost' => '/',
    ]
);

var_dump($conn->getPort()); // int(1234)
var_dump($conn->setPort(5678)); // bool(true)
var_dump($conn->getPort()); // int(1234)
```

So either `setPort` incorrectly returns `true` here, or `getPort` is broken some way, I think.

**Comment by pkruithof** [comment]
Same for AMQPQueue:

```php
$queue = new \AMQPQueue($channel);
var_dump(AMQP_IFEMPTY); // int(256)
var_dump($queue->setFlags(AMQP_IFEMPTY)); // bool(true)
var_dump($queue->getFlags()); // int(0)
```

**Comment by pinepain** [comment]
Bug with `AMQPConnection::setPort()` confirmed and should be fixed in #189.

As to `AMQPQueue` flags, from [php_amqp.h, lines 73-74](https://github.com/pdezwart/php-amqp/blob/b5f8ad6da275665dfb871e478a676969bda02d3c/php_amqp.h#L73-L74):\
> /* passive, durable, exclusive, auto-delete, no-wait (see https://www.rabbitmq.com/amqp-0-9-1-reference.html#queue.declare) */
> #define PHP_AMQP_QUEUE_FLAGS (AMQP_PASSIVE | AMQP_DURABLE | AMQP_EXCLUSIVE | AMQP_AUTODELETE)

`AMQP_IFEMPTY` and `AMQP_IFUNUSED` are not valid queue flags, though, they are valid flags for `AMQPQueue::delete()` action.

So behavior you see is expected. JFYI, by default, AMQP_AUTODELETE flag is set on new queues.

**Comment by pkruithof** [comment]
Thanks for the quick fix!
